src/opencl_common.c

```diff
@@ -88,6 +88,55 @@ extern int autotune_get_next_gws_size(size_t num, int step, int startup,
                                       int default_value);
 extern int autotune_get_prev_gws_size(size_t num, int step);
 
+/*
+ * List entry, allocated as (sizeof(struct kernel_opencl) + sizeof(data)).
+ */
+struct kernel_opencl {
+	struct kernel_opencl *next;
+
+	cl_kernel data;
+	cl_kernel *variable;
+};
```

Comment:
Why does the comment say ", allocated as (sizeof(struct kernel_opencl) + sizeof(data))" when the struct itself has a fully-sized field `data`? I think that comment should be dropped, and the allocation simplified.